docs/dg/dev/guidelines/performance-guidelines/infrastructure-worker-configuration-guidelines.md

```diff
@@ -0,0 +1,347 @@
+```
+
+**Deploy file configuration:**
+
+While FastCGI buffer sizes are handled by docker-sdk, you can configure related HTTP settings:
+
+```yaml
+# deploy.dev.yml - Configure HTTP timeout and request body size
+groups:
+  applications:
+    yves_eu:
+      application: yves
+      http:
+        timeout: 2m                    # FastCGI timeout (default: 1m)
+        max-request-body-size: 10m     # Maximum request body size (default: 1m for Yves)
+      endpoints:
+        yves.eu.spryker.local:
+          store: DE
+```
+
+**Configurable HTTP settings:**
+- `timeout`: Affects `proxy_read_timeout`, `proxy_send_timeout`, `fastcgi_read_timeout`, `fastcgi_send_timeout`
+- `max-request-body-size`: Maps to nginx `client_max_body_size`
+
+**Advanced: Buffer configuration (rarely needed)**
+
+If you need custom buffer settings (only after measuring response sizes), you can configure them at the endpoint level. Contact Spryker Cloud support before using these settings:
+
+```yaml
+# Only use after confirming need with performance testing
+endpoints:
+  yves.eu.spryker.local:
+    http:
+      buffer:
+        buffer_size: 16k           # fastcgi_buffer_size
+        buffers: 128 128k          # fastcgi_buffers
+        busy_buffers_size: 256k    # fastcgi_busy_buffers_size
+```
+
+For complete deploy file options, see:
+- [Deploy file reference - Version 1.0](https://github.com/spryker/docker-sdk/blob/master/docs/07-deploy-file/02-deploy.file.reference.v1.md)
+- [Deploy file reference (Spryker docs)](/docs/dg/dev/sdks/the-docker-sdk/deploy-file/deploy-file-reference.html)
+- [docker-sdk documentation](https://github.com/spryker/docker-sdk/tree/master/docs)
+- [Deploy file examples](https://github.com/spryker-shop/b2b-demo-shop/blob/master/deploy.dev.yml)
+
+### Measuring response sizes
+
+To determine if current buffer settings are adequate for your project:
+
+**1. Measure average response size:**
+
+```bash
+# Sample 100 requests to catalog page
+for i in {1..100}; do
+    curl -s -w "%{size_download}\n" -o /dev/null https://your-shop.com/catalog
+done | awk '{sum+=$1; count++} END {print "Average:", sum/count, "bytes"}'
+```
+
+**2. Interpret results:**
+
+- **<100KB**: Default buffer configuration (256KB) is sufficient
+- **100-500KB**: Consider requesting increased buffers via Spryker Cloud support
+- **>500KB**: Investigate response size optimization opportunities (reduce HTML/JSON payload)
+
+**3. Monitor for buffer pressure:**
+
+Look for symptoms indicating buffer exhaustion:
+- Inconsistent response times for identical requests
+- Slow responses during peak traffic
+- High disk I/O on web servers
+
+### Requesting custom nginx configuration
+
+For Spryker Cloud customers requiring custom nginx configuration:
+
+1. **Measure and document**: Provide evidence of buffer issues (response sizes, performance metrics)
+2. **Contact Spryker Cloud support**: Request configuration adjustment with justification
```

Comment:
Will the cloud really support with such requests? They will say - it's controlled via the deploy.yml, do it yourselves. Cloud does not do the edits on the customer codebase.